Counterexamples/Weierstrass.lean

```diff
@@ -0,0 +1,296 @@
+  (hasSumUniformlyOn_weierstrass ha b).summableUniformlyOn.summable (Set.mem_univ x)
+
+theorem uniformContinuous_weierstrass {a : ℝ} (ha : a ∈ Set.Ioo 0 1) (b : ℝ) :
+    UniformContinuous (weierstrass a b) := by
+  apply TendstoUniformly.uniformContinuous (tendstoUniformly_weierstrass ha b)
+  refine .of_forall fun s ↦ s.uniformContinuous_sum fun n _ ↦ ?_
+  exact (lipschitzWith_cos.uniformContinuous.comp (uniformContinuous_id.const_mul' _)).const_mul' _
+
+/-!
+### Non-differentiability
+
+To show that Weierstrass function $f(x)$ is not differentiable at any $x$, we choose a sequence
+$\{x_m\}$ such that, as $m\to\infty$
+ - $\{x_m\}$ converges to $x$
+ - The slope $(f(x_m) - f(x)) / (x_m - x)$ grows unbounded
+which means the derivative $f'(x)$ cannot exist.
+-/
+
+noncomputable
+def xm (b x : ℝ) (m : ℕ) := ⌊b ^ m * x + 3 / 2⌋ / b ^ m
+
+/-!
+Show that $x_m \in (x, x + 3 / (2b^m)]$, and it tends to $x$ by squeeze theorem.
+-/
+
+theorem lt_xm {b : ℝ} (hb : 0 < b) (x : ℝ) (m : ℕ) : x < xm b x m := by
+  unfold xm
+  grw [← Int.sub_one_lt_floor]
+  field_simp
+  grind
+
+theorem le_xm {b : ℝ} (hb : 0 < b) (x : ℝ) : (fun _ ↦ x) ≤ xm b x := fun m ↦ (lt_xm hb x m).le
+
+theorem xm_le {b : ℝ} (hb : 0 < b) (x : ℝ) : xm b x ≤ fun m ↦ x + (3 / 2) * b⁻¹ ^ m := by
+  rw [Pi.le_def]
+  intro m
+  unfold xm
+  grw [Int.floor_le]
+  apply le_of_eq
+  have hb0' : b ^ m ≠ 0 := by simp [hb.ne.symm]
+  rw [add_div, mul_div_cancel_left_of_imp (by simp [hb0']), inv_pow, div_eq_mul_inv]
+
+theorem tendsto_xm {b : ℝ} (hb : 1 < b) (x : ℝ) : Tendsto (xm b x ·) atTop (𝓝 x) := by
+  unfold xm
+  have hb0 : 0 < b := lt_trans (by norm_num) hb
+  refine tendsto_of_tendsto_of_tendsto_of_le_of_le ?_ ?_ (le_xm hb0 x) (xm_le hb0 x)
+  · exact tendsto_const_nhds_iff.mpr rfl
+  rw [show 𝓝 x = 𝓝 (x + (3 / 2) * 0) by simp]
+  refine tendsto_const_nhds.add (Tendsto.const_mul _ ?_)
+  refine tendsto_pow_atTop_nhds_zero_of_lt_one (by simpa using hb0.le) ?_
+  exact (inv_lt_one₀ hb0).mpr hb
+
+theorem tendsto_inv_xm_sub_x {b : ℝ} (hb : 1 < b) (x : ℝ) :
+    Tendsto (fun m ↦ (xm b x m - x)⁻¹) atTop atTop := by
+  have hb0 : 0 < b := lt_trans (by norm_num) hb
+  apply Tendsto.inv_tendsto_nhdsGT_zero
+  refine tendsto_nhdsWithin_iff.mpr ⟨?_, .of_forall fun m ↦ by simpa using lt_xm hb0 x m⟩
+  rw [tendsto_sub_nhds_zero_iff]
+  exact tendsto_xm hb x
+
+/-!
+To estimate the slope $(f(x_m) - f(x)) / (x_m - x)$, we break the infinite sum in
+$f(x_m) - f(x)$ into two parts
+
+$$ f(x_m) - f(x) = ∑_{n=0}^{m-1} a^n (\cos(b^n\pi x_m) - \cos(b^n\pi x))
++ ∑_{n=m}^{\infty} a^n (\cos(b^n\pi x_m) - \cos(b^n\pi x)) = A + B$$
+-/
+
+/-- The partial sum has upper bound in absolute value $|A| \le |x_m - x| \pi (ab)^m / (ab - 1)$ -/
+theorem weierstrass_partial {a : ℝ} (ha : 0 < a) {b : ℕ} (hab : 1 < a * b) (x : ℝ) (m : ℕ) :
+    |∑ n ∈ Finset.range m, a ^ n * (cos (b ^ n * π * xm b x m) - cos (b ^ n * π * x))| ≤
+    |xm b x m - x| * (π / (a * b - 1) * (a * b) ^ m) := by
```

Comment:
because this is a term continued from the previous line, it gets extra indentation.
```suggestion
      |xm b x m - x| * (π / (a * b - 1) * (a * b) ^ m) := by
```